### PR TITLE
Add name for Edge

### PR DESCRIPTION
**Pull request by Linary**
* name is concated by sort values

Change-Id: I2eaa7ecd0e2d6dbd587063b466908b554f8cbbcd

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/driver/GraphManager.java [hunk 1]
```diff
@@ -55,6 +58,10 @@ public String graph() {
         return this.graph;
     }
 
+    public SchemaManager schemaManager() {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/driver/GraphManager.java:61]: rename to schema()

File: src/main/java/com/baidu/hugegraph/structure/graph/Edge.java [hunk 2]
```diff
@@ -138,6 +147,32 @@ public void targetLabel(String targetLabel) {
...
+        SchemaManager schemaManager = this.manager.schemaManager();
+        List<String> sortKeys = schemaManager.getEdgeLabel(this.label)
+                                             .sortKeys();
+        if (sortKeys.isEmpty()) {
+            return ImmutableList.of();
+        }
+        List<Object> propValues = new ArrayList<>(sortKeys.size());
+        for (String sortKey : sortKeys) {
+            // The call frequency is too high
+            PropertyKey propertyKey = schemaManager.getPropertyKey(sortKey);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/structure/graph/Edge.java:167]: can we cache it

File: src/main/java/com/baidu/hugegraph/structure/graph/Edge.java [hunk 3]
```diff
@@ -138,6 +147,32 @@ public void targetLabel(String targetLabel) {
         this.targetLabel = targetLabel;
     }
 
+    public String name() {
+        if (this.name == null) {
+            this.name = SplicingIdGenerator.concatValues(this.sortValues());
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/structure/graph/Edge.java:152]: can just extract from id?

Comment (Linary) [reply src/main/java/com/baidu/hugegraph/structure/graph/Edge.java:152]: yes

File: pom.xml [hunk 4]
```diff
@@ -126,6 +126,8 @@
                 <configuration>
                     <excludes>
                         <exclude>com/baidu/hugegraph/example/*.class</exclude>
+                        <exclude>com/baidu/hugegraph/util/IdUtil.class</exclude>
+                        <exclude>com/baidu/hugegraph/util/SplicingIdGenerator.class</exclude>
```
Comment (javeme) [inline pom.xml:130]: also copy the unit test

Comment (Linary) [reply pom.xml:130]: Only exist IdUtilTest.class in core

Comment (javeme) [reply pom.xml:129]: add test for SplicingIdGenerator, also add to hugegraph-server

File: src/test/java/com/baidu/hugegraph/unit/UnitTestSuite.java [hunk 5]
```diff
@@ -30,7 +30,8 @@
     BatchElementRequestTest.class,
     PropertyKeyTest.class,
     IndexLabelTest.class,
-    CommonUtilTest.class
+    CommonUtilTest.class,
+    SplicingIdGeneratorTest.class
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/unit/UnitTestSuite.java:34]: add IdUtilTest

File: src/test/java/com/baidu/hugegraph/functional/EdgeTest.java [hunk 6]
```diff
@@ -685,15 +705,9 @@ public void testIterateEdgesByVertexId() {
...
-            Iterator<Edge> iter = graph().iterateEdges(markoId, Direction.OUT,
-                                                       "knows", properties,
-                                                       1);
-            Iterators.size(iter);
-        }, e -> {
-            Assert.assertEquals("Can't query by paging and filtering",
-                                e.getMessage());
-        });
+        Iterator<Edge> iter = graph().iterateEdges(markoId, Direction.OUT,
+                                                   "knows", properties, 1);
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/functional/EdgeTest.java:709]: add case for "query by paging and filtering""

**Review by javeme — APPROVED** [review]

**Review by houzhizhen — APPROVED** [review]

**Pull request merged**